src/substrait/textplan/converter/PlanPrinterVisitor.cpp

```diff
@@ -7,58 +7,567 @@
+      "List selects are not yet supported: " + select.ShortDebugString());
+  return std::string("LISTSELECT_NOT_YET_IMPLEMENTED");
+}
+
+std::any PlanPrinterVisitor::visitListSelectItem(
+    const ::substrait::proto::
+        Expression_MaskExpression_ListSelect_ListSelectItem& item) {
+  error_listener_->addError(
+      "List select items are not yet supported: " + item.ShortDebugString());
+  return std::string("LISTSELECTITEM_NOT_YET_IMPLEMENTED");
+}
+
+std::any PlanPrinterVisitor::visitMapSelect(
+    const ::substrait::proto::Expression_MaskExpression_MapSelect& select) {
+  error_listener_->addError(
+      "Map selects are not yet supported: " + select.ShortDebugString());
+  return std::string("MAPSELECT_NOT_YET_IMPLEMENTED");
+}
+
+std::any PlanPrinterVisitor::visitExpressionLiteralStruct(
+    const ::substrait::proto::Expression_Literal_Struct& structure) {
+  error_listener_->addError(
+      "Expression literals are not yet supported: " +
+      structure.ShortDebugString());
+  return std::string("EXPRLIT_NOT_YET_IMPLEMENTED");
+}
+
+std::any PlanPrinterVisitor::visitFileOrFiles(
+    const ::substrait::proto::ReadRel_LocalFiles_FileOrFiles& structure) {
+  error_listener_->addError(
+      "Local file or files is not yet supported: " +
+      structure.ShortDebugString());
+  return std::string("FORF_NOT_YET_IMPLEMENTED");
+}
+
 std::any PlanPrinterVisitor::visitAggregateFunction(
     const ::substrait::proto::AggregateFunction& function) {
-  return std::string("AF-NOT-YET-IMPLEMENTED");
+  std::stringstream text;
+  text << lookupFunctionReference(function.function_reference()) << "(";
+  bool hasPreviousText = false;
+  for (const auto& arg : function.arguments()) {
+    if (hasPreviousText)
+      text << ", ";
+    switch (arg.arg_type_case()) {
+      case ::substrait::proto::FunctionArgument::kEnum:
+        text << "ENUM_NOT_SUPPORTED";
+        break;
+      case ::substrait::proto::FunctionArgument::kType:
+        text << ANY_CAST(std::string, visitType(arg.type()));
+        break;
+      case ::substrait::proto::FunctionArgument::kValue:
+        text << ANY_CAST(std::string, visitExpression(arg.value()));
+        break;
+      case ::substrait::proto::FunctionArgument::ARG_TYPE_NOT_SET:
+      default:
+        text << "UNKNOWN_AGGREGATE_FUNCTION_ARGUMENT_TYPE";
+        break;
+    }
+    hasPreviousText = true;
+  }
+  for (const auto& arg : function.args()) {
+    if (hasPreviousText)
+      text << ", ";
+    text << ANY_CAST(std::string, visitExpression(arg));
+    hasPreviousText = true;
+  }
+  text << ")";
+  for (const auto& option : function.options()) {
+    text << "#" << option.name();
+    for (const auto& pref : option.preference()) {
+      text << ";" << pref;
+    }
+  }
+  text << "->" << ANY_CAST(std::string, visitType(function.output_type()));
+  // TODO -- Emit the requested sort behavior here.
+  text << "@" << ::substrait::proto::AggregationPhase_Name(function.phase());
```

Comment:
Bleh...I think this will be a little wordy.  Maybe some shorthand here at some point? Also, we can probably default / leave blank if it is `AGGREGATION_PHASE_INITIAL_TO_RESULT`.  Although the docs say the default is `INTERMEDIATE_TO_RESULT` :frowning_face:

Reply:
Agreed this part of the textplan is harder to consume, especially if we have to use the full enum names.  Added to the list.